tests/RouterTest.php

```diff
@@ -10,28 +10,221 @@ public function notfound()
         }
     }
 
+    class Test3
+    {
+        public static function test()
+        {
+            echo 'Test3 hello';
+
+            return false;
+        }
+    }
+
     class RouterTest extends PHPUnit_Framework_TestCase
     {
-        protected function setUp()
+        public function testInit()
         {
-            // Clear SCRIPT_NAME because bramus/router tries to guess the subfolder the script is run in
-            $_SERVER['SCRIPT_NAME'] = '/index.php';
+            $this->assertInstanceOf('\Bramus\Router\Router', new \Bramus\Router\Router());
+        }
 
-            // Default request method to GET
-            $_SERVER['REQUEST_METHOD'] = 'GET';
+        public function testCloneRouter()
+        {
+            $_SERVER['REQUEST_URI'] = '/novel/info?p1=1';
 
-            // Default SERVER_PROTOCOL method to HTTP/1.1
-            $_SERVER['SERVER_PROTOCOL'] = 'HTTP/1.1';
+            // Create Router instance
+            $router = new \Bramus\Router\Router();
+
+            $router2 = clone $router;
+
+            ob_start();
+            // Define routes
+            $router2->prefix('/novel')->get('/info', function () {
```

Comment:
Does named functions also are still working? eg.: `->get('/info', 'MyNamespace\MyController@method')`

Reply:
Would also something like this work?

```
function ($router2) {
    $router2->get('/info', function () {};

    ....
}($router2->prefix('/novel'));

Reply:
Maybe something like:
```
$router2->prefix('/novel')->group($router) {}
```

Would be cool, what you think?